### PR TITLE
Add auth0 example

### PR DESCRIPTION
**Pull request by timneutkens**
After talking to @luisrudge I decided to update the example to reflect changes in 2.0 and move it into the next.js repo so it can be easily found.

### CODE REVIEW

File: examples/with-auth-auth0/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+now
+```
+
+## The idea behind the example
+
+This example shows how to implement authentication in next.js using Auth0
+
+## Credits
+
+This example was origionally built by [@luisrudge](https://github.com/luisrudge) here: [luisrudge/next.js-auth0](https://github.com/luisrudge/next.js-auth0)
```
Comment (sedubois) [inline examples/with-auth-auth0/README.md]: origionally

**Comment by sedubois** [comment]
I'm a bit of a newb with auth, but is it safe to use cookies that aren't httpOnly?

See https://github.com/zeit/next.js/issues/153#issuecomment-264215442

**Comment by timneutkens** [comment]
https://github.com/zeit/next.js/issues/153#issuecomment-272393862

Going to have a look at @iaincollins's starter setup 👍

**Comment by sedubois** [comment]
@timneutkens you can also look at my project for ideas: 
https://github.com/relatenow/relate/

I took some ideas from @iaincollins' project back then. Would be interested in any feedback.

**Comment by thangngoc89** [comment]
AFAIK, there are 2 places that store the jwt token here: 

- localStorage
- and cookie 

- I know that cookie is required for next.js server to render a page properly when user is authenticated. If there is a need for gathering info from external API, the server will request the API on behalf of user (using the jwt token from cookie). But this mean the site is now vulnerable for CSRF attack isn't it? 

- Also storing jwt token in localStorage or cookie would make the site vulnerable for XSS attack. 

Could you explain a little these things @timneutkens ? I'm new to authentication mechanisms.

**Comment by timneutkens** [comment]
@thangngoc89 cc @luisrudge

**Comment by timneutkens** [comment]
Closing this one in favor of Iain's work which is more secure.

**Comment by thangngoc89** [comment]
@timneutkens could you drop a link to lain's work?

**Comment by timneutkens** [comment]
https://github.com/iaincollins/nextjs-starter

**Comment by thangngoc89** [comment]
@timneutkens thank you

**Pull request closed without merging**